File: SquirrelsNest.Almond/src/app/UserData/user.data.reducer.ts
import {Action} from '@ngrx/store'
import {CLEAR_USER_DATA, UPDATE_USER_DATA, UpdateUserData} from './user.data.actions'
import {initialUserDataState, UserDataState} from './user.data.state'

export function userDataReducer( state: UserDataState = initialUserDataState, action: Action ): UserDataState {
  switch( action.type ) {
    case UPDATE_USER_DATA:
      const updateUserData = action as UpdateUserData

      return {
        lastProject: updateUserData.userData.currentProject,
        displayStyle: updateUserData.userData.displayStyle,
        displayOnlyMyIssues: updateUserData.userData.displayOnlyMyIssues,
        displayCompletedIssues: updateUserData.userData.displayCompletedIssues,
      }

    case CLEAR_USER_DATA:
      return {
        lastProject: initialUserDataState.lastProject,
        displayStyle: initialUserDataState.displayStyle,
        displayOnlyMyIssues: initialUserDataState.displayOnlyMyIssues,
        displayCompletedIssues: initialUserDataState.displayCompletedIssues,
      }

    default:
      return state
  }
}

File: SquirrelsNest.Pecan/Client/UserData/Reducers/UserDataReducer.cs
using Fluxor;
using SquirrelsNest.Pecan.Client.UserData.Actions;
using SquirrelsNest.Pecan.Client.UserData.Store;

namespace SquirrelsNest.Pecan.Client.UserData.Reducers {
    // ReSharper disable once UnusedType.Global
    public static class UserDataReducer {
        [ReducerMethod]
        public static UserDataState UpdateUserData( UserDataState state, RequestUserDataSuccess action ) {
            var listStyle = IssueListStyle.FullDetail;

            if(( action.UserData.IssueListDisplayStyle.Equals( IssueListStyle.TitleOnly )) ||
               ( action.UserData.IssueListDisplayStyle.Equals( IssueListStyle.TitleDescription )) ||
               ( action.UserData.IssueListDisplayStyle.Equals( IssueListStyle.FullDetail ))) {
                listStyle = action.UserData.IssueListDisplayStyle;
            }

            return new( action.UserData.CurrentProjectId,
                        action.UserData.DisplayCompletedIssues, 
                        action.UserData.DisplayCompletedIssuesLast,
                        action.UserData.DisplayOnlyMyAssignedIssues,
                        listStyle );
        }
            
    }
}
